catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -581,29 +583,63 @@ public Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
   @Override
   public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmptySchemaException {
     try {
+      Namespace filesetNs =
+          Namespace.of(
+              Stream.concat(Stream.of(ident.namespace().levels()), Stream.of(ident.name()))
+                  .toArray(String[]::new));
+      List<FilesetEntity> filesets =
+          store.list(filesetNs, FilesetEntity.class, Entity.EntityType.FILESET);
+      if (filesets.size() > 0 && !cascade) {
+        throw new NonEmptySchemaException("Schema %s is not empty", ident);
+      }
+
+      // Delete all the managed filesets no matter whether the storage location is under the
+      // schema path or not.
+      // The reason why we delete the managed fileset's storage location one by one is because we
+      // may mis-delete the storage location of the external fileset if it happens to be under
+      // the schema path.
+      filesets.stream()
+          .filter(f -> f.filesetType() == Fileset.Type.MANAGED)
+          .forEach(
+              f -> {
+                try {
+                  Path filesetPath = new Path(f.storageLocation());
+                  FileSystem fs = getFileSystem(filesetPath, conf);
+                  if (fs.exists(filesetPath)) {
+                    if (!fs.delete(filesetPath, true)) {
+                      LOG.warn("Failed to delete fileset {} location {}", f.name(), filesetPath);
+                    }
+                  }
+                } catch (IOException ioe) {
+                  LOG.warn(
```

Comment:
I'm afraid there will be some issues as the following:
- Due to some unknown reasons, a few of the files under the schema can't be removed. In this case, should we ignore it?
- Can we remove files one by one and then drop the schema in a transaction?

Reply:
> Due to some unknown reasons, a few of the files under the schema can't be removed. In this case, should we ignore it?

Do you have better suggestions? If some files cannot be deleted, we just have the warning log and remove other fileset locations as possible as we can. I don't think throwing an exception instead is better compared to this.

> Can we remove files one by one and then drop the schema in a transaction?

I don't think it is achievable, right?